scripts/precalculate.js

```diff
@@ -329,6 +331,40 @@ function main() {
 		}
 	}
 
+	// Calculate optimised polestars
+	let polestarCombos = {};
+	for (let crew of crewlist) {
+		if (!crew.in_portal) continue;
+		let polestars = crew.traits.slice();
+		polestars.push('crew_max_rarity_'+crew.max_rarity);
+		for (let skill in crew.base_skills) {
+			if (crew.base_skills[skill]) polestars.push(skill);
+		}
+		let onePolestarCombos = polestars.slice().map((pol) => [pol]);
+		let twoPolestarCombos = _.combinations(polestars, 2);
+		let threePolestarCombos = _.combinations(polestars, 3);
+		let fourPolestarCombos = _.combinations(polestars, 4);
+		let crewPolestarCombos = [].concat(onePolestarCombos).concat(twoPolestarCombos).concat(threePolestarCombos).concat(fourPolestarCombos);
+		for (let combo of crewPolestarCombos) {
+			let sorted = combo.sort();
+			if (!polestarCombos[sorted]) {
+				polestarCombos[sorted] = {
+					count: 0,
+					crew: [],
+					polestars: sorted,
+				}
+			}
+			polestarCombos[sorted].count = polestarCombos[sorted].count + 1;
+			polestarCombos[sorted].crew.push(crew.symbol);
+		}
+	}
+
+	for (let crew of crewlist) {
+		if (!crew.in_portal) continue;
+		let uniqueCombos = Object.keys(polestarCombos).filter((pc) => polestarCombos[pc].count === 1 && polestarCombos[pc].crew[0] === crew.symbol).map((pc) => polestarCombos[pc].polestars);
+		crew.unique_polestar_combos = uniqueCombos;
+	}
```

Comment:
```suggestion
		let comboIds = [];	// Potential list of combos to check later
		for (let combo of crewPolestarCombos) {
			let sorted = combo.sort();
			if (!polestarCombos[sorted]) {
				polestarCombos[sorted] = {
					count: 0,
					crew: [],
					polestars: sorted,
				}
				// Only add new combos to list; if it already exists in polestarCombos,
				//	its count is > 1 and is of no use to us here
				comboIds.push(sorted);
			}
			polestarCombos[sorted].count = polestarCombos[sorted].count + 1;
			polestarCombos[sorted].crew.push(crew.symbol);
		}
		crew._comboIds = comboIds;	// Attach as temp property
	}

	for (let crew of crewlist) {
		if (!crew.in_portal) continue;
		let uniqueCombos = [];
		// Now double check a crew's list of combos to find counts that are still 1
		crew._comboIds.forEach((pc) => {
			if (polestarCombos[pc].count === 1)
				uniqueCombos.push(polestarCombos[pc].polestars);
		});
		crew.unique_polestar_combos = uniqueCombos;
		delete crew._comboIds;	// Don't need it anymore
	}
```

Reply:
Here's the logic:
Two loops as before. First loop is mostly the same, with this addition: For each crew, assign a potential list of combo IDs to check later. This list only includes brand new entries to full polestarCombos list, anything else we can assume is not a unique combo.

On the second crew loop, iterate through each crew's combo IDs list (instead of filtering the entire polestarCombos array) and collect the unique combos.